### ISSUE TITLE
[flake8-bugbear] Avoid false positive for usage `return None` (B901)

### ISSUE THREAD

**Issue by f4rk4sh** [astral-sh/ruff#12156]
A minimal code snippet that reproduces the bug is almost the same as [here](https://github.com/astral-sh/ruff/commit/312f6640b837eb1189748980707cfed70d1f5ffb#diff-3da9dc4c996fc465b80db3e323994049105aa08d73a56f50a4991481408f90e4R16) but with explicit `return None`:

```python
def not_broken():
        if True:
            return None

        yield 3
        yield 2
        yield 1
```

The false positive:

```sh
B901 Using `yield` and `return {value}` in a generator function can lead to confusing behavior
   |
64 |     def not_broken():
65 |         if True:
66 |             return None
   |             ^^^^^^^^^^^ B901
67 | 
68 |         yield 3
   |

Found 1 error.
```

The command invoked:

```yaml
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.5.0
    hooks:
      - id: ruff
        args: ["--fix", "--exit-non-zero-on-fix", "--preview"]
```

**Comment by charliermarsh** [comment]
Can you remind me, why is this _not_ a problem?

**Comment by charliermarsh** [comment]
Thank you. I was mostly trying to remember why the implicit `return` is considered ok in the first place.

**Comment by zanieb** [comment]
I think the idea is that `return` is a valid way to stop iteration

> In Python 3 return {value} can be used in a generator; however, the combination of yield and return can lead to confusing behavior, as the return statement will cause the generator to raise StopIteration with the value provided, rather than returning the value to the caller.

I actually think the rule is fine here, i.e.,`return None` doesn't feel like a false positive because it is misleading as the rule is intended to flag.

**Comment by charliermarsh** [comment]
Ah thank you. I think I agree -- this seems like a reasonable thing to flag.
